### ISSUE TITLE
How to create files with contents upto size mentioned?

### ISSUE THREAD

**Issue by SamPosh** [Microsoft/diskspd#51]
I have to fill my disk with files upto reserved quota and see the eventvwr for warning. 
To fill the data i am trying to use fsutil createnew <file> <size> but it created sparse files.

I tried with Vdbench. It created files with content but it throws exception at end. 

When i was searching i found that diskspd is IO tool for Windows so i thought of using it .
I tried with Diskspd below command to fill the disk. 

`.\diskspd.exe -c7G -b4K -t2 -w100 C:\UnMapTest-Volume1\disk1\testfile.dat`

But it also create only Sparse file without any content.
Whether Diskspd can be used for filling the random data in disk? If so how to create files in the folder with content upto some GB (say 7GB)

**Comment by lauracaulfield** [comment]
Hi Sam,

The cmd you're running probably doesn't run for long enough.  Check out the -d flag.  It specifies the runtime, and defaults to 10s.  This won't be long enough to write 7GB of data.

**Comment by dl2n** [comment]
Laura is correct.

However, you should have seen that the first portion of the file was written with a pattern of incrementing bytes similar to what I’m showing below, as seen in a hex editor.

This is DISKSPD’s default fill pattern. If you want another one, either fully random or sampled from an existing file source, see the -Z switch.

 00000000  00 01 02 03 04 05 06 07-08 09 0A 0B 0C 0D 0E 0F  *................*
00000010  10 11 12 13 14 15 16 17-18 19 1A 1B 1C 1D 1E 1F  *................*
00000020  20 21 22 23 24 25 26 27-28 29 2A 2B 2C 2D 2E 2F  * !"#$%&'()*+,-./*
00000030  30 31 32 33 34 35 36 37-38 39 3A 3B 3C 3D 3E 3F  *0123456789:;<=>?*
00000040  40 41 42 43 44 45 46 47-48 49 4A 4B 4C 4D 4E 4F  *@ABCDEFGHIJKLMNO*
00000050  50 51 52 53 54 55 56 57-58 59 5A 5B 5C 5D 5E 5F  *PQRSTUVWXYZ[\]^_*
00000060  60 61 62 63 64 65 66 67-68 69 6A 6B 6C 6D 6E 6F  *`abcdefghijklmno*
00000070  70 71 72 73 74 75 76 77-78 79 7A 7B 7C 7D 7E 7F  *pqrstuvwxyz{|}~.*
00000080  80 81 82 83 84 85 86 87-88 89 8A 8B 8C 8D 8E 8F  *................*

Second, you should have received the following:

   WARNING: target access pattern will not be sequential, consider -si

The default sequential pattern (also specified with -s) has the threads you specified (-t2) each starting at offset zero of the file and proceeding forward independently, racing on top of each other to write at offset zero, 4K, 8K, and so forth. Interlocked sequential has them share a single incrementing offset which allows them to cooperatively issue exactly one IO for offset 0, 4K, 8K, and so forth up the file. This is probably closer to what you intended, hence the WARNING. Another option, depending on your goals, would be to separate the starting offset of the threads with -T.

Sent from Mail<https://go.microsoft.com/fwlink/?LinkId=550986> for Windows 10

________________________________

**Comment by SamPosh** [comment]
@lauracaulfield  and @dl2n   I am very new to Diskspd (even for storage domain). I have seen in Vdbench an option for stopafter some GB. So it will not run based on duration. Is there any option for running Diskspd till it fill the required size rather than mention the timing? Because it is not easier to calculate the time it takes for filling the required size.

**Comment by lauracaulfield** [comment]
Dan can correct me if I'm wrong, but I don't believe there is a way to stop after a certain amount of data.  However, you can estimate the runtime you will need from Diskspd's output.  When you use the text output (the default), look for the total MB/s under the Write IO text block.  (row is labeled "total" column is labeled "MB/s")  If it reports 120MB/s it will take 8s to write 1GB.  (1024MB / 120MB/s)

Another option is the really bare-bones tool in StorScore, here: https://github.com/Microsoft/StorScore/tree/master/src/precondition  

Here is an example of how it's used:
https://github.com/Microsoft/StorScore/blob/master/lib/PreconditionRunner.pm

**Comment by dl2n** [comment]
That’s correct; at this time there is no way to focus on writing a fixed amount.

Sent from Mail<https://go.microsoft.com/fwlink/?LinkId=550986> for Windows 10

________________________________
